docs/design/storagenode-graceful-exit.md

```diff
@@ -91,6 +92,133 @@ This process including the Storage nodes transferring their pieces to other node
 	- Receiving storage nodes should be uploaded to as if they were receiving any other upload (orders, etc).
 - We need to have the storage node keep track how its progress during graceful exit by calculating how much data it had when it started the graceful exit and keeping track of how much data it has deleted (data is deleted from the exiting node as it sends pieces to new nodes). This is an estimation since the storage node is likely holding more data than it will gracefully exit (garbage data). 
 
+## Implementation (MVP)
+
+#### Satellite
+- Update DBX - Add updateable `exit_initiated_dt` and `exit_completed_dt` timestamps to nodes table with indexes
+  - Add GetExitingNodeIds method to overlaycache. Returns nodes IDs where `exit_initiated_dt` is not null and `exit_completed_dt` is null.
+- Create GracefulExit endpoint
+  - Endpoints should be secured using the peer Identity provided in context
+  - Initiates the exit by setting `nodes.exit_initiated` to current time
+  - ``` 
+	service GracefulExit {
+		rpc Initiate(stream InitiateRequest) returns (stream InitiateResponse) {}
+	}
+
+	message InitiateRequest {
+	}
+
+	message InitiateResponse {
+	}
+	```
+- Update DBX - Add table exit_pieceinfos
+-  ```
+	model exit_pieceinfo (
+		key node_id path
+
+		field node_id           blob
+		field path              blob
+		field peice_info        blob
+		field durability_ratio  float64
+		field queued_dt         timestamp ( autoinsert )
+		field sent_dt           timestamp ( updateable )
+		field completed_dt      timestamp ( updateable )
+	)
+   ```
+- Add `PieceAction` field to, `cache.FindStorageNodesRequest`. Update `cache.FindStorageNodesWithPreferences` to ignore exiting nodes for uploads and repairs.
```

Comment:
instead of piece action, can u query to find nodes that have only 'exit_initiated_dt' null?, that way no need to add a new piece action.

Reply:
The new PieceAction field indeed seems unnecessary. The caller can take advantage of the existing `ExcludedNodes` field to list the current nodes from the pointer.

Reply:
Agreed.  Not sure where I was going with this one.  Document updated.